### ISSUE TITLE
Cache size not detected on AMD Rome machines

### ISSUE THREAD

**Issue by jgarvin** [bemeurer/cache-size#5]
I am running:

                (1..u8::MAX).cartesian_product([cache_size::CacheType::Data, cache_size::CacheType::Unified].iter())
                    .filter_map(|(n, t)| cache_size::cache_size(n, *t))
                    .reduce(|a,b| max(a,b))
                    .expect("Could not determine largest cache size!")

And getting the panic. But I'm pretty sure there are caches on an AMD Rome :)

Here is `/proc/cpuinfo` for one core:

```
processor	: 0
vendor_id	: AuthenticAMD
cpu family	: 23
model		: 49
model name	: AMD EPYC 7H12 64-Core Processor
stepping	: 0
microcode	: 0x8301034
cpu MHz		: 3295.550
cache size	: 512 KB
physical id	: 0
siblings	: 64
core id		: 0
cpu cores	: 64
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 16
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate sme ssbd mba sev ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif umip rdpid overflow_recov succor smca
bugs		: sysret_ss_attrs spectre_v1 spectre_v2 spec_store_bypass
bogomips	: 5199.91
TLB size	: 3072 4K pages
clflush size	: 64
cache_alignment	: 64
address sizes	: 43 bits physical, 48 bits virtual
power management: ts ttp tm hwpstate cpb eff_freq_ro [13] [14]
```

**Comment by lovesegfault** [comment]
I have a Ryzen box, let me see if I can reproduce. If I'm unable to I might have a hard time debugging this as I don't have access to any EPYC machines.

**Comment by lovesegfault** [comment]
I can reproduce on my Ryzen box. But I suspect the issue comes from `raw_cpuid`.

**Comment by lovesegfault** [comment]
This is the related issue: https://github.com/gz/rust-cpuid/issues/26

**Comment by lovesegfault** [comment]
I'm working on this since the upstream issue was closed. It's nontrivial though.

**Comment by daniel-brenot** [comment]
I've added an update in a PR if you feel like merging it. Tested on a friends desktop and it seems to give reliable results, so let me know what you think. 😄

**Comment by daniel-brenot** [comment]
@lovesegfault Bumping this because it would be nice to have the patch merged and published :)

**Comment by lovesegfault** [comment]
Should be fixed and released in v0.6.0.

Thanks @daniel-brenot for the work on fixing this!

My Zen machine is currently somewhere in a moving truck, I'm going to defer closing this until someone can confirm the issue is indeed gone :)

**Comment by lovesegfault** [comment]
Alright, no one has commented on this in a while, so I'll be closing it.

Thanks, everyone, for the help in fixing it!
